### ISSUE TITLE
Unreliable torch

### ISSUE THREAD

**Issue by nitricware** [twostraws/CodeScanner#85]
Tested on an iPhone 11 Pro running iOS 16;

With `isTorchOn` set to `true` the torch can show three behaviours - none of which is reliable:

1. The torch will not light up
2. The torch will light up for a brief moment and then be off again.
3. The torch will not go out when the View is dismissed.

The View is presented in a Sheet:

```swift
.sheet(isPresented: $isShowingScanner) {
    CodeScannerView(
        codeTypes: [.code39],
        simulatedData: "04 ASO SCANTEST123",
        isTorchOn: true,
        completion: handleScan)
}
```

**Comment by NathanFallet** [comment]
Can you provide a project (for example a zip archive) to reproduce this issue? I tried in my app and it is not happening...

**Comment by nitricware** [comment]
Yes, it happens in this project: https://github.com/nitricware/pSTaRT

**Comment by ClementCardonnel** [comment]
I'm experiencing this issue as well. This seems to be related to these stack overflow questions:

- [AVCaptureSession freezes when torch is turned on](https://stackoverflow.com/questions/52297812/avcapturesession-freezes-when-torch-is-turned-on)
- [Device torch turns off when starting AVCaptureSession](https://stackoverflow.com/questions/53657378/device-torch-turns-off-when-starting-avcapturesession)

**Comment by NathanFallet** [comment]
Maybe related to #70

**Comment by marckrenn** [comment]
I think the problem boils down to these LOCs [here](https://github.com/twostraws/CodeScanner/blob/65cd76bf88532eb4bfeaeb08981e87907e5f8da4/Sources/CodeScanner/ScannerViewController.swift#LL135C13-L135C78)

`let fallbackVideoCaptureDevice = AVCaptureDevice.default(for: .video)`

and [here](https://github.com/twostraws/CodeScanner/blob/main/Sources/CodeScanner/ScannerViewController.swift#L395)

`if let backCamera = AVCaptureDevice.default(for: AVMediaType.video),`

So the quickfix is to use the exact same `AVCaptureDevice` on both lines, the proper fix would be to pass explicitly set `AVCaptureDevice` to the `updateViewController`-function, I think.

**Comment by nitricware** [comment]
I changed my code, so that the `CodeScannerView` gets passed a `@State` and the torch is off initially. I also implemented a button as suggested in #70 but when I press the button the torch switches on as expected, however the camera freezes.
